docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+
+:::
+
+## Set the Retry Policy
+
+Temporal makes software durable and fault-tolerant by design.
+The Service can "retry" Activities, which are failable by nature.
+That is, it can wait a little time for an issue to resolve, and then try running the Activity again, repeating as needed.
+You customize how this happens in a retry policy.
+That policy, along with time-out settings, is defined in the following Workflow implementation:
+
+<!--SNIPSTART money-transfer-java-workflow-implementation {"selectedLines": ["12-37"]}-->
 <!--SNIPEND-->
 
-Task Queues are defined by a simple string name.
+This code includes several components:
+
+* **`RetryOptions`** define how to automatically re-attempt an Activity that's failed.
+  This includes backoff coefficients (`setBackoffCoefficient`) that treat service providers' endpoints with respect, so you don't slam them at once.
+  Exponential backoffs allow a service time to recover.
+  In this code, if an Activity fails, the Service waits a second, then two seconds, then four, etc.
+  It fails after five attempts (`setMaximumAttempts`).
+  It won't wait more than 20 seconds between attempts (`setMaximumInterval`).
+* **`ActivityOptions`** define fixed limits for how long an Activity can keep trying before the Service shuts down further attempts.
+  *Always* set a schedule-to-close timeout limit.
+  Schedule-to-close includes the entire duration of an activity.
+  Otherwise, by default, Temporal retries failed activities forever.
+* **Activity-specific options** adjust options for specific Activity types.
+
+:::note Retryable Errors
+
+Although not shown in this code sample, you can specify Activity errors that Temporal should not attempt to retry.
+Set the Retry Option's non-retryable errors field.
+This provides you with the flexibility to choose the circumstances to retry.
+
+:::
+
+Now that you've seen the code, it's time to look at the Workflow in action.
+In the next sections, you'll launch the Temporal Server and create a Workflow Execution.
+
+## Start the Workflow
+
+In this section, you'll run and observe a Workflow using the project code and the Temporal Web UI.
+
 
-<!--SNIPSTART money-transfer-project-template-java-shared-constants-->
-[src/main/java/moneytransferapp/Shared.java](https://github.com/temporalio/money-transfer-project-template-java/blob/4e1ca12d8d1c058c67d810efe8dcf68f7b318208/src/main/java/moneytransferapp/Shared.java)
-```java
-public interface Shared {
+### Prepare to run
 
-    static final String MONEY_TRANSFER_TASK_QUEUE = "MONEY_TRANSFER_TASK_QUEUE";
-}
+Start running a [Temporal Service](https://docs.temporal.io/clusters) by launching it from a Terminal session, as discussed in the [Getting Started tutorial](https://learn.temporal.io/getting_started/java/dev_environment/):
+
+
+```bash
+temporal server start-dev
```

Comment:
We decided a while back to always set the UI Port to 8080. People default look for it there and if the cli team ever decides to move ports again (it's happened) then it doesn't affect any of our conten.t 
```suggestion
temporal server start-dev --ui-port 8080
```